cmake/extensions.cmake

```diff
@@ -79,26 +79,28 @@ endfunction()
 
 # https://cmake.org/cmake/help/latest/command/target_include_directories.html
 function(zephyr_include_directories)
-  foreach(arg ${ARGV})
-    if(IS_ABSOLUTE ${arg})
-      set(path ${arg})
-    else()
-      set(path ${CMAKE_CURRENT_SOURCE_DIR}/${arg})
-    endif()
-    target_include_directories(zephyr_interface INTERFACE ${path})
+  foreach(relative_path ${ARGV})
+    __maybe_convert_from_relative_to_absolute_path(${relative_path} absolute_path)
+    target_include_directories(zephyr_interface INTERFACE ${absolute_path})
   endforeach()
 endfunction()
 
 # https://cmake.org/cmake/help/latest/command/target_include_directories.html
 function(zephyr_system_include_directories)
-  foreach(arg ${ARGV})
-    if(IS_ABSOLUTE ${arg})
-      set(path ${arg})
+  foreach(relative_path ${ARGV})
+    __maybe_convert_from_relative_to_absolute_path(${relative_path} absolute_path)
+    target_include_directories(zephyr_interface SYSTEM INTERFACE ${absolute_path})
+  endforeach()
+endfunction()
+
+function(__maybe_convert_from_relative_to_absolute_path input output)
```

Comment:
we should not use `_` as prefix for functions.

The `_` is used by CMake to prepend existing functions when they are overloaded, as can be seen by this example:
```
# test.cmake
function(A)
  message("I'm in A")
endfunction()

function(A)
  message("I'm in B")
endfunction()

A()
_A()
```
and running this:
```
$ cmake -P test.cmake
I'm in B
I'm in A
```

Just make a comment and add private to its name, like in this case:
https://github.com/zephyrproject-rtos/zephyr/blob/25e6803705b25602eb0b730a03a254c354bfae9e/cmake/extensions.cmake#L2986-L2987